src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +750,82 @@ class C1<T>
             Assert.Equal(c1.MangleName, c1r.MangleName);
             Assert.Equal(c1.MetadataName, c1r.MetadataName);
         }
+
+        [Fact]
+        public void RetargetingFunctionPointerType_01()
+        {
+            string source = @"
+unsafe class C
+{
+    delegate*<out string, ref readonly int> M(delegate* cdecl<void> param) => throw null;
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.UnsafeReleaseDll);
+            comp.VerifyDiagnostics();
+
+            var sourceModule = comp.SourceModule;
+            var sourceAssembly = (SourceAssemblySymbol)sourceModule.ContainingAssembly;
+            var sourceNamespace = sourceModule.GlobalNamespace;
+
+            var retargetingAssembly = new RetargetingAssemblySymbol(sourceAssembly, isLinked: false);
+            retargetingAssembly.SetCorLibrary(sourceAssembly.CorLibrary);
+            var retargetingModule = retargetingAssembly.Modules[0];
+            var retargetingNamespace = retargetingModule.GlobalNamespace;
+
+            RetargetingSymbolChecker.CheckSymbols(sourceNamespace.GetMember<NamedTypeSymbol>("C"), retargetingNamespace.GetMember<NamedTypeSymbol>("C"));
```

Comment:
>RetargetingSymbolChecker.CheckSymbols( [](start = 12, length = 38)

I have no idea what this this thing is doing and why do we need to call it and fix it up.

Reply:
If the goal is to test that the function pointer symbol is not getting modified in the retargeting process, I recommend to use approach taken by the test below. Only don't change mscorelib version. Change the version of some other reference, could be any other library, even custom one. 

---
In reply to: [414901658](https://github.com/dotnet/roslyn/pull/43625#discussion_r414901658) [](ancestors = 414901658)